src/compartment_sets.cpp

```diff
@@ -151,6 +151,22 @@ class CompartmentSet {
         compartment_locations_.reserve(comp_it->size());
         for (auto&& el : *comp_it) {
             compartment_locations_.emplace_back(CompartmentSet::_parseCompartmentLocation(el));
+            if (compartment_locations_.size() >= 2) {
+                const auto& prev = compartment_locations_[compartment_locations_.size() - 2];
+                const auto& curr = compartment_locations_.back();
+                if (curr <= prev) {
+                    throw SonataError(
+                        fmt::format("CompartmentSet 'compartment_set' must be strictly sorted. "
```

Comment:
Looks good, but I think this error isn't as helpful as it could be for duplicates:
```
----> 1 libsonata.CompartmentSets('''
      2 {
      3   "example_compartment_set": {
      4     "population": "S1nonbarrel_neurons",
      5     "compartment_set": [
      6       [0, 2, 0.3],
      7       [0, 2, 0.3]
      8     ]
      9   }
     10 }
     11 ''')

SonataError: CompartmentSet 'compartment_set' must be strictly sorted. Found CompartmentLocation(0, 2, 0.3) after CompartmentLocation(0, 2, 0.3)
```
Could duplicates be flagged as duplicates?